src/main/java/core/basesyntax/service/impl/DataValidatorServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.service.impl;
+
+import static core.basesyntax.model.FruitTransaction.greatOperationPattern;
+
+import core.basesyntax.service.DataValidatorService;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class DataValidatorServiceImpl implements DataValidatorService {
+    private static final Pattern INTEGER_PATTERN = Pattern.compile("-?\\d+");
+    private static final Pattern OPERATION_PATTERN = greatOperationPattern();
+    private static final String SEPARATOR = ",";
+    private static final int INDEX_SEPARATOR = 0;
+    private static final int INDEX_FOR_OPERATION_IN_STRING = 0;
+    private static final int INDEX_FOR_PRODUCT_NAME_IN_STRING = 1;
+    private static final int INDEX_FOR_PRODUCT_VALUE_IN_STRING = 2;
+    private static final int CORRECT_NUMBERS_OF_COMMAS = 2;
+    private static final int INDEX_CORRECT_FOR_LINE_IN_WHICH_ERROR = 2;
+    private static final int HEADER_INDEX = 0;
+
+    @Override
```

Comment:
Consider renaming method 'isDataValid' to 'validateData' and changing its return type to void, since it throws exceptions when data is invalid and doesn't return false.